File: backend/src/main/java/com/tmk/vtcmanager/application/domain/operation/CategorieOperation.java
package com.tmk.vtcmanager.application.domain.operation;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CategorieOperation {

    private Long id;
    private String code;
    private String libelle;
    private TypeOperation typeOperation;
    private NatureResultat natureResultat;
    /** Compte du plan comptable (SYSCOHADA) pour l'export ; null si non mappé. */
    private String compteComptable;
    private boolean actif;
    private SousCategorieOperation sousCategorie;
}

File: mobile/lib/features/operation_financiere/domain/entities/categorie_operation.dart
import '../enums/type_operation.dart';
import 'sous_categorie_operation.dart';

class CategorieOperation {
  final int id;
  final String code;
  final String libelle;
  final TypeOperation typeOperation;
  final bool actif;
  final SousCategorieOperation? sousCategorie;

  const CategorieOperation({
    required this.id,
    required this.code,
    required this.libelle,
    required this.typeOperation,
    required this.actif,
    this.sousCategorie,
  });
}
